source/MaterialXGenShader/Nodes/HwCompoundNode.h

```diff
@@ -0,0 +1,26 @@
+#ifndef MATERIALX_HWCOMPOUNDNODE_H
+#define MATERIALX_HWCOMPOUNDNODE_H
+
+#include <MaterialXGenShader/Nodes/CompoundNode.h>
+#include <MaterialXGenShader/HwShaderGenerator.h>
+
+namespace MaterialX
+{
+
+/// Extending the CompoundNode with requirements for HW.
```

Comment:
It's because for HW we need to deal with closure contexts, emitting up to three different versions of the node function if the node represents a BSDF (reflection, transmission and indirect). Previously this context handling was done in the general CompountNode version, but since it's only needed for HW I made a class specialization for it.